src/main/kotlin/gg/essential/elementa/components/Window.kt

```diff
@@ -175,6 +175,11 @@ class Window @JvmOverloads constructor(
     override fun mouseClick(mouseX: Double, mouseY: Double, button: Int) {
         requireMainThread()
 
+        //  Override mouse positions to be in the center of the pixel on Elementa versions
+        //  2 and over. See [ElementaVersion.V2] for more info.
+        val (mouseX, mouseY) = super.transformMouseLocation(mouseX.toFloat(), mouseY.toFloat()).let { (x, y) ->
+            x.toDouble() to y.toDouble()
+        }
```

Comment:
This is shadowing the existing `mouseX` and `mouseZ` variable, which should generally be avoided (at least IntelliJ is emitting a warning for it, and not getting into the habit of ignoring warnings should be reason enough to not do it). Now, not shadowing would be risky as well, cause then you could accidentally use the wrong one, but there's another way which has neither of the downsides:
```kotlin
        //  Override mouse positions to be in the center of the pixel on Elementa versions 2 and over.
        //  See [ElementaVersion.V2] for more info.
        val (adjustedX, adjustedY) = transformMouseLocation(mouseX, mouseY)

        doMouseClick(adjustedX, adjustedY, button)
    }

    private fun doMouseClick(mouseX: Double, mouseY: Double, button: Int) {
```
Also, the `super.` is redundant.
Also, I'd change that method to operate on `Double`s because most related things already do. If you inline `UIComponent.getMouseX/Y`, then literally the only thing using `Float` is the return type of `getMousePosition`. And this way we're safe from any potential loss of precision due to the cast to `Float`.